components/scrubber/scrubber.go

```diff
@@ -94,11 +106,26 @@ func (s *scrubberImpl) JSON(msg json.RawMessage) (json.RawMessage, error) {
 
 // KeyValue implements Scrubber
 func (s *scrubberImpl) KeyValue(key string, value string) (sanitisedValue string) {
-	f, ok := s.FieldIndex[strings.ToLower(key)]
-	if !ok {
+	sanitisatiser := s.GetSanitisatiser(key)
+	if sanitisatiser == nil {
 		return value
 	}
-	return f(value)
+	return sanitisatiser(value)
+}
+
+// GetSanitisatiser implements
+func (s *scrubberImpl) GetSanitisatiser(key string) Sanitisatiser {
+	if f, ok := s.FieldIndex[key]; ok {
+		return f
+	}
+	for regexp, f := range s.RegexpIndex {
+		if regexp.MatchString(key) {
+			s.FieldIndex[key] = f
+			return f
+		}
+	}
+	s.FieldIndex[key] = SanitizeNoop
```

Comment:
How big will this map grow? How do we make sure we're not bursting at the seams here, if e.g. we feed this with an instance ID as key?

Reply:
I won't expect it should happen in normal usage, hope noone is putting uuids as keys. But I put a limit of 1Mb on len of all keys and added simple eviction algorithm which delete least frequent keys in batches till it gets enough memory.